### DISCUSSION TITLE
How quartz trigger jobs ?

### DISCUSSION THREAD

**Discussion by juliendu11** [quartznet/quartznet#1214] [Q&A] [answered]
Hello,

I have a question about how Quartz works

If I have multiple recurring jobs like:

- Every Monday at 2 p.m.
- Every Monday at 4 p.m.
- Every Tuesday at 11 a.m.
- Every Wednesday at 11 a.m.
....

How does quartz work? How are jobs triggered?

Does it launch a scan every x time and if the date and time corresponds to the job it triggers it

Or does it program the jobs with a timer at each change of day (if we have just gone from Sunday to Monday, it looks at the jobs available for today and starts a timer for x time to start the works (timer for 2 p.m. and timer for 4 p.m.))

Or other ?

In fact I would like to know how quartz works inside?

I have a project where I have to schedule a lot of recurring work by days and hours

thank you in advance

**Comment by lahma** [comment, marked as answer]
In a nutshell, each trigger has next fire time which is calculated on first store and after each fire. So we have a bunch of datetimes and we choose ones that are near enough to be fired and execute them in their prioritized order obeying concurrency rules. This scanning is done by a background thread which then dispatches the work, job invocation, to thread pool.

**Reply by juliendu11** [reply]
Ok, thank you for your reply
